### ISSUE TITLE
Version 0.3.5 fails as a result of upgrading 'globby' from v9.2.0 to v10.0.1

### ISSUE THREAD

**Issue by barakman** [OpenZeppelin/solidity-docgen#73]
Specifically, at file [compile.ts](https://github.com/OpenZeppelin/solidity-docgen/blob/master/src/compile.ts), the following line:
```
const files = await filter.files();
```
Which is translated in file compile.js to:
```
const files = yield globby_1.default(path_1.default.join(directory, '**/*.sol'), {
    ignore: ignore.map(i => path_1.default.join(i, '**/*')),
});
```
The output `files` is an empty array.

I believe that this is due to breaking changes in the `globby` package, between v9.2.0 and v10.0.0.

See https://github.com/sindresorhus/globby/releases/tag/v10.0.0.

Thanks

**Comment by frangio** [comment]
Hi @barakman, thanks for reporting. We're not seeing this error in OpenZeppelin's docs. I've also cloned [bancorprotocol/contracts](https://github.com/bancorprotocol/contracts) and updated `solidity-docgen` to `0.3.5` and it seems to pick up all contracts, I didn't get an empty `files` array.

Can you share steps to reproduce the issue?

**Comment by barakman** [comment]
@frangio: Hi, thank you for the quick response.

Command line:
```
node node_modules/solidity-docgen/dist/cli.js --input=solidity/contracts --output=solidity/docgen/docs --templates=solidity/docgen --solc-module=node_modules/truffle/node_modules/solc --solc-settings={"optimizer":{"enabled":true,"runs":200}} --contract-pages
```
Printout:
```
TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at SolcAdapter.compile (C:/Users/Temp/Desktop/Bancor/contracts/node_modules/solidity-docgen/dist/compile.js:78:41)
    at Object.<anonymous> (C:/Users/....../node_modules/solidity-docgen/dist/compile.js:51:33)
    at Generator.next (<anonymous>)
    at fulfilled (C:/Users/....../node_modules/solidity-docgen/dist/compile.js:5:58)
```
Thanks

**Comment by barakman** [comment]
The `master` branch or the `0.5.4` branch - same results.

But have you made sure to update in file `package.json` from 0.3.4 to 0.3.5?

**Comment by frangio** [comment]
Yes, by running `npm install solidity-docgen@0.3.5`.

**Comment by barakman** [comment]
Well, that's awfully strange, on my side it clearly fails.

The differences between us, I would imagine, are node/npm versions, globally installed packages (I only have `windows-build-tools` on my platform), and perhaps the OS itself (I'm on Windows 10).

My versions are:
- node version 10.16.0
- npm version 6.9.0

But I sincerely doubt that it is either one of the above, because the process completes successfully with `solidity-docgen` 0.3.4, while failing with `solidity-docgen` 0.3.5.

Any ideas what could be causing it?

Thanks

**Comment by barakman** [comment]
In addition to that, I was able to deterministically track it down to the line:
```
const files = yield globby_1.default(path_1.default.join(directory, '**/*.sol'), {
    ignore: ignore.map(i => path_1.default.join(i, '**/*')),
});
```
In file `compile.js`.

And I have verified that the following inputs to this line are identical for both 0.3.4 and 0.3.5:
- `path_1.default.join(directory, '**/*.sol')`
- `ignore.map(i => path_1.default.join(i, '**/*'))`

The difference was in the output of that line, which is an array of file names in 0.3.4, an an empty array in 0.3.5.

This difference is clearly the result of the implementation function `globby_1.default`, because the input to this function was identical in both cases.

When I checked `npm ls globby`, I got 9.2.0 on `solidity-docgen` 0.3.4, and 10.0.1 on `solidity-docgen` 0.3.5.

When I then checked the release notes for `globby` 10.0.0 (at https://github.com/sindresorhus/globby/releases/tag/v10.0.0), I noticed that they've explicitly mentioned API-breaking changes.

All of this leads me to the conclusion that the problem is within this area.

That said, the fact that it runs smoothly on your end kinda refutes my conclusion...

**Comment by frangio** [comment]
It must be the difference in OS. I would roll back this update but I feel like we should get to the bottom of the issue and fix it instead.

I don't see anything in the list of `globby` or `fast-glob` breaking changes that could be affecting this. Do you see anything that could be related?

Can you try manually running `globby` yourself in a Node console using the same inputs that docgen is using? Perhaps that will give us more data to narrow down the issue.

**Comment by barakman** [comment]
Version 0.3.4 1st input: `solidity\contracts\**\*.sol`
Version 0.3.5 1st input: `solidity\contracts\**\*.sol`

Version 0.3.4 2nd input: `{ ignore: [] }`
Version 0.3.5 2nd input: `{ ignore: [] }`

Version 0.3.4 output: an array of file names
Version 0.3.5 output: `[]`

If your OS-conjecture is true, then it might be worth noting that in version 0.3.4 output, all file names are given with normal (forward) slashes.
I'm saying this in conjunction with the fact that the 1st input is given with back-slashes (in both versions), and Windows paths are typically also given with back-slashes.
Not sure if it's a good lead though...

**Comment by frangio** [comment]
Right. One of the breaking changes in `fast-glob` says:
>Only forward-slashes in glob expression. Previously, we convert all slashes to the forward-slashes, which did not allow the use of escaping. See pattern syntax section in the README.md file.

Can you try replacing `path_1.default` with `path_1.posix` here:

```javascript
const files = yield globby_1.default(path_1.default.join(directory, '**/*.sol'), {
    ignore: ignore.map(i => path_1.default.join(i, '**/*')),
});
```

So it would become:

```javascript
const files = yield globby_1.default(path_1.posix.join(directory, '**/*.sol'), {
    ignore: ignore.map(i => path_1.posix.join(i, '**/*')),
});
```

Taken from [Windows vs POSIX in Node.js `path`](https://nodejs.org/api/path.html#path_windows_vs_posix).

**Comment by barakman** [comment]
Good catch! One sec...

**Comment by frangio** [comment]
`path` is one of the default Node.js modules. I may have given wrong instructions because that's TypeScript compiled code.

If you run `require('path').posix.join('a', 'b')` in a Node.js console it should return `'a/b'`.

**Comment by barakman** [comment]
Yeah, I just realized that (see updated comment above).

In file `compile.js` I see:
```
const path_1 = __importDefault(require("path"));
```

Would that yield the exact same effect as what you've just suggested?

**Comment by barakman** [comment]
Ahh, it won't. The `__importDefault` makes the difference (I'm guessing by this symbol's name).

**Comment by frangio** [comment]
Awesome, I'll publish a fix in a bit. However, I'll publish 0.3.7. Hopefully there won't be any surprises for you in the 0.3.7 update. (0.3.6 was skipped.)

**Comment by barakman** [comment]
@frangio:
OK, I'll wait for 0.3.8.
Thanks again for your help :)

**Comment by frangio** [comment]
Fixed in https://github.com/OpenZeppelin/solidity-docgen/commit/6fae9f1b2c335f91c1ac7ba81042c1ebe36bbadf and released in 0.3.8!
